learn/appchains/tx-fees.md

```diff
@@ -0,0 +1,116 @@
+
+- **Type 0 (Legacy)** - the transaction fee is calculated through a single _gasPrice_ value that is included in the signed transaction blob. Because Tanssi EVM-compatible Appchains have a dynamic pricing mechanism, _gasPrice_ must be greater than the current block _baseFee_ for a transaction to be considered valid
+- **Type 2 (EIP-1559)** - the transaction fee is calculated with a combination of the _maxFeePerGas_ and _maxPriorityFeePerGas_ from signed transaction blob, and the network's _baseFee_ dynamically changes based on block congestion
+
+Independently of the transaction type, the outcome of all EVM transactions is that there is an associated cost in native tokens that the network must charge. 
+
+By default, Tanssi EVM-compatible Appchains are configured with the following parameters:
+
+- **Minimum BaseFee** - the minimum gas price of the network in case there are no transactions for long periods. The default value is set to 1 _GWei_
+- **Block Fulness Target (Elasticity)** - the target gas used in a block so that the _baseFee_ remains the same. [EIP-1559](https://eips.ethereum.org/EIPS/eip-1559){target=\_blank} defines this value as a constant set to 2, meaning that the target usage is 50% of the block gas limit. All Tanssi EVM-compatible Appchains are set with the same target
+- **Maximum BaseFee Increase** - the maximum amount the _baseFee_ can increase or decrease, in percent points, based on the previous block target usage. [EIP-1559](https://eips.ethereum.org/EIPS/eip-1559){target=\_blank} defines this value as a constant set to 12.5\%. Consequently, if the block is full/empty, the _baseFee_ will increase/decrease by 12.5\%, and any intermediate values are linearly adjusted. Developers can configure this value for Tanssi EVM-compatible Appchains, but the default value is 12.5\%
+
+!!! note
+    One key difference in Tanssi EVM-compatible Appchains EIP-1559 implementation is that the transaction fees are calculated using the previous block _baseFee_.
+
+The EVM transaction fee cost associated with all Tanssi EVM-compatible Appchains is captured at an EVM execution level. Nevertheless, EVM transactions do take block execution time. Therefore a _GasToWeight_ algorithm is required to account for the weight consumed by a specific call relative to the gas it is consuming.
+
+<!-- https://github.com/moondance-labs/tanssi/blob/master/container-chains/templates/frontier/runtime/src/lib.rs#L825 -->
+<!-- https://github.com/polkadot-evm/frontier/blob/272fe8839f87161ed89350de166b379f1f4c6136/primitives/evm/src/lib.rs#L253-L265 -->
+
+Ultimately, the transaction fee and weight associated to an EVM call in a Tanssi EVM-compatible Appchain can be calculated with the following formula:
+
+=== "EIP-1559"
+    ```
```

Comment:
you can use text as the langauge for the ``` back ticks

Reply:
Thanks I did not know that!